Replicator/tests/ReplicatorAPITest.cc

```diff
@@ -613,6 +613,215 @@ TEST_CASE_METHOD(ReplicatorAPITest, "Is Document Pending", "[C][Push]") {
+    // This won't actually be replicated, but is needed to call c4repl_getPendingDocIDs for the collection
+    repl::C4ReplParamsCollection paramsFederation { Federation };
+    paramsFederation.push = kC4OneShot;
+    paramsFederation.pull = kC4Disabled;
+    paramsCouncil.callbackContext = this;
+    paramsCouncil.socketFactory = _socketFactory;
+
+    int expectedPending = 0;
+    SECTION("Normal") {
+        expectedPending = 100;
+    }
+
+    SECTION("Filtered") {
+        expectedPending = 99;
+        paramsCouncil.replCollection.callbackContext = this;
+        paramsCouncil.pushFilter = []( C4CollectionSpec collectionSpec, C4String docID, C4String revID,
+                                C4RevisionFlags flags, FLDict flbody, void *context) {
+            auto test = (ReplicatorAPITest*)context;
+            c4repl_getStatus(test->_repl);
+            return FLSlice_Compare(docID, "0000005"_sl) != 0;
+        };
+    }
+
+    SECTION("Set Doc IDs") {
+        expectedPending = 2;
+        FLEncoder e = FLEncoder_New();
+        FLEncoder_BeginDict(e, 1);
+        FLEncoder_WriteKey(e, FLSTR(kC4ReplicatorOptionDocIDs));
+        FLEncoder_BeginArray(e, 2);
+        FLEncoder_WriteString(e, FLSTR("0000002"));
+        FLEncoder_WriteString(e, FLSTR("0000004"));
+        FLEncoder_EndArray(e);
+        FLEncoder_EndDict(e);
+        options = FLEncoder_Finish(e, nullptr);
+        paramsCouncil.replCollection.optionsDictFleece = C4Slice(options);
+        FLEncoder_Free(e);
+    }
+
+    _repl = c4repl_newLocal(db, (C4Database*)db2, paramsCouncil, ERROR_INFO(err));
+    REQUIRE(_repl);
+
+    // Not actually used for replication
+    auto replFed = c4repl_newLocal(db, (C4Database*)db2, paramsFederation, ERROR_INFO(err));
+    REQUIRE(replFed);
+
+    FLSliceResult_Release(options);
+
+    // Check that collection 1 has the right amount of pending documents
+    C4SliceResult encodedDocIDs = c4repl_getPendingDocIDs(_repl, Council, ERROR_INFO(err));
+    CHECK(err.code == 0);
+    REQUIRE(encodedDocIDs != nullslice);
+    FLArray docIDs = FLValue_AsArray(FLValue_FromData(C4Slice(encodedDocIDs), kFLTrusted));
+    CHECK(FLArray_Count(docIDs) == expectedPending);
+    c4slice_free(encodedDocIDs);
+    
+    // Replicate collection 1
+    c4repl_start(_repl, false);
+    REQUIRE_BEFORE(5s, c4repl_getStatus(_repl).level == kC4Stopped);
+
+    // Now collection 1 shouldn't have any pending documents
+    encodedDocIDs = c4repl_getPendingDocIDs(_repl, Council, ERROR_INFO(err));
+    CHECK(err.code == 0);
+    REQUIRE(encodedDocIDs == nullslice);
+
+    // Check that collection 2 still has all the documents pending
+    encodedDocIDs = c4repl_getPendingDocIDs(replFed, Federation, ERROR_INFO(err));
+    CHECK(err.code == 0);
+    REQUIRE(encodedDocIDs != nullslice);
+    docIDs = FLValue_AsArray(FLValue_FromData(C4Slice(encodedDocIDs), kFLTrusted));
+    CHECK(FLArray_Count(docIDs) == 100);
+
+    c4repl_free(replFed);
```

Comment:
Can probably use c4::ref to be safer here (it will wrap the return of c4repl_newLocal)